src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx

```diff
@@ -104,18 +104,14 @@ export class FieldsBreakdownScene extends SceneObjectBase<FieldsBreakdownSceneSt
   };
 
   private serviceFieldsChanged = (newState: ServiceSceneState, oldState: ServiceSceneState) => {
-    if (JSON.stringify(newState.fields) !== JSON.stringify(oldState.fields)) {
+    if (JSON.stringify(newState.fields) !== JSON.stringify(oldState.fields) || newState.loading !== oldState.loading) {
```

Comment:
Ohhhh, right. More reasons to move away from this type of checks.

Reply:
Yeah, it's like a useEffect dependency, but there's no eslint rule to let you know when something is referenced but not in the dep array

Reply:
What would happen if the fields are the same, but the loading state changed? I think the "or" will completely override this check.

Reply:
should be fine, it'll just update the loading state in that case

Reply:
But yeah, we'll call `getLabelOptions` uncessarily, splitting this out into two functions, for updating loading state, and updating options might be better